From 43c9eb08a86391e5651fa27ba676af8150673593 Mon Sep 17 00:00:00 2001
From: Liwei Song <liwei.song@windriver.com>
Date: Wed, 18 May 2022 17:41:14 +0800
Subject: [PATCH] include: configs: soc64: set correct QSPI clock for Agilex
 plaform

upstream kernel commit d2e593084270 ("arm64: dts: intel: socfpga_agilex:
move clocks out of soc node") has move qspi-clock out of soc node,
adjust linux_qspi_enable command accordingly.

Upstream-Status: Submitted (https://github.com/altera-opensource/u-boot-socfpga/pull/8)

Signed-off-by: Liwei Song <liwei.song@windriver.com>
---
 include/configs/socfpga_soc64_common.h | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/include/configs/socfpga_soc64_common.h b/include/configs/socfpga_soc64_common.h
index ce6a1f634a89..30cb9fd363e2 100755
--- a/include/configs/socfpga_soc64_common.h
+++ b/include/configs/socfpga_soc64_common.h
@@ -170,8 +170,10 @@ unsigned int cm_get_qspi_controller_clk_hz(void);
 		"fdt set /soc/spi@ff8d2000 status okay;" \
 		"if fdt set /soc/clocks/qspi-clk clock-frequency" \
 		" ${qspi_clock}; then" \
+		" else if fdt set /clocks/qspi-clk clock-frequency" \
+		" ${qspi_clock}; then" \
 		" else fdt set /soc/clkmgr/clocks/qspi_clk clock-frequency" \
-		" ${qspi_clock}; fi; fi\0" \
+		" ${qspi_clock}; fi; fi; fi\0" \
 	"scriptaddr=0x05FF0000\0" \
 	"scriptfile=boot.scr\0" \
 	"nandroot=ubi0:rootfs\0" \
@@ -224,8 +226,10 @@ unsigned int cm_get_qspi_controller_clk_hz(void);
 		"fdt set /soc/spi@ff8d2000 status okay;" \
 		"if fdt set /soc/clocks/qspi-clk clock-frequency" \
 		" ${qspi_clock}; then" \
+		" else if fdt set /clocks/qspi-clk clock-frequency" \
+		" ${qspi_clock}; then" \
 		" else fdt set /soc/clkmgr/clocks/qspi_clk clock-frequency" \
-		" ${qspi_clock}; fi; fi\0" \
+		" ${qspi_clock}; fi; fi; fi\0" \
 	"scriptaddr=0x02100000\0" \
 	"scriptfile=boot.scr\0" \
 	"fatscript=if fatload mmc 0:1 ${scriptaddr} ${scriptfile};" \
-- 
2.17.1

